### PR TITLE
[Mono] Intrinsify Vector.Equals

### PR DESCRIPTION
**Pull request by fanyang-mono**
Contribute to the list of #64072

This change enables SIMD intrinsics replacement for `Vector.Equals` on amd64 with and without LLVM.

### CODE REVIEW

**Comment by SamMonoRT** [comment]
@simonrozsival @mkhamoyan @steveisok fyi

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -866,6 +867,16 @@ emit_sys_numerics_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSigna
 	case SN_ConvertToInt64:
 	case SN_ConvertToUInt32:
 	case SN_ConvertToUInt64:
+		break;
+	case SN_Equals:
+		MonoType *type;
```
Comment (vargaz) [inline src/mono/mono/mini/simd-intrinsics.c]: Needs a {} block

Comment (fanyang-mono) [reply src/mono/mono/mini/simd-intrinsics.c]: I moved the declaration out of the switch statement in the second commit. Do you think it is better to keep the declaration inside the switch statement?

**Comment by fanyang-mono** [comment]
I am currently focusing on implementing Arm64 intrinsics. Will close this PR for now, and will circle back to it once the arm64 work is done.

**Pull request closed without merging**